crates/hyperdrive-math/src/short/close.rs

```diff
@@ -418,4 +477,68 @@ mod tests {
         assert!(result.is_err());
         Ok(())
     }
+
+    // Tests market valuation against hyperdrive valuation when closing a short.
+    // This function aims to give an estimated position value without considering
+    // slippage, market impact, or any other liquidity constraints. As such, its
+    // divergence with the hyperdrive valuation will grow under low liquidity
+    // conditions. For this reason, we relax the error tolerance in such cases.
+    #[tokio::test]
+    async fn test_calculate_market_value_short() -> Result<()> {
+        let tolerance = int256!(1e12); // 0.000001
+
+        // Fuzz the spot valuation and hyperdrive valuation against each other.
+        let mut rng = thread_rng();
+        for _ in 0..*FAST_FUZZ_RUNS {
+            let mut scaled_tolerance = tolerance;
+
+            let state = rng.gen::<State>();
+            let bond_amount = state.minimum_transaction_amount();
+            let open_vault_share_price = rng.gen_range(fixed!(0.5e18)..=fixed!(2.5e18));
+            let maturity_time = U256::try_from(state.position_duration())?;
+            let current_time = rng.gen_range(fixed!(0)..=FixedPoint::from(maturity_time));
+
+            // When the reserves ratio is too small, the market impact makes the error between
+            // the valuations larger, so we scale the test's tolerance up to make up for it,
+            // since this is meant to be an estimate that ignores liquidity constraints.
+            let reserves_ratio = state.effective_share_reserves()? / state.bond_reserves();
+            if reserves_ratio < fixed!(1e12) {
```

Comment:
I think you want to nest these, unless the intention is to have the scale be 1,000 if it is <1e14? If so, I'd be more explicit. So either

```
if reserves_ratio >= fixed!(1e14) && reserves_ratio < fixed!(1e12) {
    scaled_tolerance *= int256!(100); // 1e14
}
else if reserves_ratio < fixed!(1e14) {
    scaled_tolerance *= int256!(10); // 1e13
}
```

or

```
if reserves_ratio >= fixed!(1e14) && reserves_ratio < fixed!(1e12) {
    scaled_tolerance *= int256!(100); // 1e14
}
else if reserves_ratio < fixed!(1e14) {
    scaled_tolerance *= int256!(1_000); // 1e15
}
```

Reply:
You're absolutely right! They were meant to be nested. Excellent catch, pushing it now, merging once CI passes!